subworkflows/local/prepare_input_stitch/prepare_input_stitch.nf

```diff
@@ -0,0 +1,65 @@
+workflow PREPARE_INPUT_STITCH {
+
+    take:
+    ch_posfile
+    ch_fasta
+    ch_input_impute
+
+    main:
+
+    ch_versions      = Channel.empty()
+
+    // Value channels
+    def input_empty         = [[]]
+    def rdata_empty         = [[]]
+    k_val                   = params.k_val
+    ngen                    = params.ngen
+
+    if  (params.panel && params.step.split(',').contains("panelprep")) {
+        // Get chromosomes of posfile
+        ch_posfile = ch_posfile.map{meta, posfile -> return[['chr': meta.chr], posfile]}
+
+        // Get chromosomes of fasta
+        ch_chromosomes = ch_fasta.map{it -> it[2]}
+                        .splitCsv(header: ["chr", "size", "offset", "lidebase", "linewidth", "qualoffset"], sep: "\t")
+                        .map{it -> return [[chr: it.chr], it.chr]}
+
+        // Make final channel with parameters
+        stitch_parameters = ch_posfile.map { it + input_empty + rdata_empty}
+                        .join(ch_chromosomes)
+                        .map { it + k_val + ngen}
+    } else if (params.posfile){
+        // Get unique chromosomes in posfile
+        ch_chromosomes_posfile = ch_posfile.map{it -> it[1]}
+                .splitCsv(header: ["chr", "pos", "ref", "alt"], sep: "\t")
+                .map{it -> return [it.chr]}
+
+        ch_chromosomes_posfile = ch_chromosomes_posfile.unique()
+
+        // Make final channel with parameters
+        stitch_parameters = ch_posfile.map { it + input_empty + rdata_empty}
```

Comment:
Isn't ch_posfile already separated by chromosomes ?
If so it should have the "chr" inside its meta map, not ?